### DISCUSSION TITLE
Need feeback about results and some concepts explanations

### DISCUSSION THREAD

**Discussion by jej** [kernc/backtesting.py#306] [Q&A]
Hi,
I am starting to test backtesting.py, coming form freqtrade world. I am a bit confused about the results I get. Probably because I am wrong on some concepts. The whole domain is quite new for me (trading, algo trading, etc).
My first test is on a crypto. The strategy is really stupid, just to test the behaviour. Buying if price under a value and selling if above a greater value. I would expect it makes positive return when the conditions are good, but it's not the case. Here are the data.

OHLCV in a csv file:
```
DateTime,Open,High,Low,Close,Volume
2021-01-01 01:00:00,37.4121,37.4258,37.3931,37.416,91.89
2021-01-01 01:01:00,37.4231,37.4707,37.4049,37.4362,309.48
2021-01-01 01:02:00,37.4448,37.446,37.3686,37.3969,591.96
2021-01-01 01:03:00,37.3847,37.3847,37.3398,37.3398,391.06
2021-01-01 01:04:00,37.3381,37.3581,37.316,37.3314,216.2
2021-01-01 01:05:00,37.32,37.32,37.2082,37.2276,1816.21
2021-01-01 01:06:00,37.2469,37.2524,37.1696,37.1919,344.9
.../...
```
Strategy:
```
from backtesting import Strategy

class Dummy(Strategy):
    def init(self):
        pass
    
    def next(self):
        if self.data.Close < 248:
            self.position.close()
            self.buy()
        elif self.data.Close > 255:
            self.position.close()
            self.sell()

from backtesting import Backtest

bt = Backtest(DATA, Dummy, cash=1000, commission=.002, exclusive_orders=True)
stats = bt.run()
stats

bt.plot(plot_volume=False)

```
Stats:
```
Start                     2021-03-01 00:00:00
End                       2021-03-02 00:00:00
Duration                      1 days 00:00:00
Exposure Time [%]                   99.861207
Equity Final [$]                   272.927655
Equity Peak [$]                   1004.482297
Return [%]                         -72.707234
Buy & Hold Return [%]               21.439925
Return (Ann.) [%]                         0.0
Volatility (Ann.) [%]                     NaN
Sharpe Ratio                              NaN
Sortino Ratio                             NaN
Calmar Ratio                              0.0
Max. Drawdown [%]                  -73.465468
Avg. Drawdown [%]                  -36.743712
Max. Drawdown Duration        0 days 23:55:00
Avg. Drawdown Duration        0 days 11:59:00
# Trades                                 1033
Win Rate [%]                        22.749274
Best Trade [%]                       3.412652
Worst Trade [%]                     -1.079074
Avg. Trade [%]                      -0.162712
Max. Trade Duration           0 days 01:09:00
Avg. Trade Duration           0 days 00:02:00
Profit Factor                        0.254888
Expectancy [%]                       -0.16205
SQN                                -15.576835
_strategy                               Dummy
_equity_curve                             ...
_trades                         Size  Entr...
```

The plot:
![image](https://user-images.githubusercontent.com/33821/114286119-846ea300-9a4b-11eb-94a7-3fc5bbfa22c0.png)

As the market start very low and cross several times my limits, as I buy lower than I sell, how is it possible to get a negative return at the end, in the favourable chosen conditions?
I would expect to have blank trade periods, when it's sold above 255, until a buy occurs again under 248. Here the train lines are continuous from beginning to end.
I think too I am misunderstanding the short/long meanings, regarding buying/selling actions. This is weird for me:
```
            self.position.close()
            self.buy()
```
If someone has a good definition... I got some other information elsewhere in the trading web sphere, but it makes me more confused! I don't know if the concepts here are related to general trading/backtesting functioning, or related to specific backtesting.py internals/constrains.
I am surprised by the count of trades (1033) in one day. I guess these are only my buy/sell signals. I would like to know how many effective buys/sells occurred in that day.
Well, if someone has a bit of time to explain, welcome!
Thanks a lot.

**Comment by kernc** [comment]
>         if self.data.Close < 248:
>             self.position.close()
>             self.buy()

This matches and runs (position closing and re-buying) on _every_ bar of your example. If you're looking for a cross-over, see [`backtesting.lib.crossover()`](https://kernc.github.io/backtesting.py/doc/backtesting/lib.html#backtesting.lib.crossover).

**Reply by jej** [reply]
Thanks for your answer. It's a bit more logical now with the crossover. Still not understand why the trades train lines are continuous, even on periods with no active trades.
Regards.
